the-collab-lab/tcl-73-smart-shopping-list
# Add dark mode toggle

## Description

This code adds a toggle switch for dark mode and light mode.

This came about as I was working on a problem in my previous issue where the body of the app turned white while the header and nav stayed dark, after implementing CSS changes for the sign in alerts. I figured I'd give it a go!

### Update
App now recognizes users system preference and will set the theme accordingly. It will also automatically change the theme if you change your system preference in your computer settings. AC and Testing Steps have been updated.

## Related Issue

N/A

## Acceptance Criteria
- [x] User can toggle between a dark mode and a light mode by clicking on the switch.
- [x] Users theme preference persists after refreshing the page.
- [x] App theme is initially set to the users system preference.
- [x] Changing theme preference in macOS/Windows settings automatically changes theme in the app.

## Type of Changes

Enhancement, Accessibility

## Updates

### Before

![image](https://github.com/the-collab-lab/tcl-73-smart-shopping-list/assets/78065152/cedfda85-21cd-4415-beb5-65ef32b0269d)

### After
**Switch added for toggling dark mode and light mode**
![image](https://github.com/the-collab-lab/tcl-73-smart-shopping-list/assets/78065152/0cbb8273-b332-46b2-99ec-8bf6a10ac9c3)
![image](https://github.com/the-collab-lab/tcl-73-smart-shopping-list/assets/78065152/1c4f4476-9db3-445b-9a25-36dcda6906c8)



## Testing Steps / QA Criteria
`git pull`
`git checkout ja-feature-dark-mode-toggle`
`npm ci`
`npm start`

- Upon viewing the app for the first time, the theme should match your system preference by default.
- Changing your system preference in macOS or Windows settings should automatically update the theme on the app.
- Click on the toggle switch on the upper right corner to see if it correctly toggles between dark mode and light mode.
- Choose light mode and refresh the page, then navigate between pages. The light mode preference should persist.
- Choose dark mode and refresh the page, then navigate between pages. The dark mode preference should persist.

## Notes
- ~~Right now this does not automatically set the theme based on the users system/browser preference as it did before. There are methods to detect the users preference using JS and the CSS media queries we currently have, if that's a functionality we want to implement.~~
